src/scene/scene_lua.inl

```diff
@@ -896,4 +899,41 @@ int worldRotation(lua_State* L) {
     return 1;
 }
 
+/**
+ * \ingroup LuaScripts
+ * convertStringToLuaAndPush_bool(lua_State*, string):
+ * Used to convert a string value from a text file (e.g. profile) to a lua bool type.
+ * If string value is a bool, the value gets pushed to the lua_State, and returns true.
+ * If string value is not a bool, the lua_State is unchanged and returns false.
+ */
+bool convertStringToLuaAndPush_bool(lua_State* L, std::string s) {
```

Comment:
Not sure if they need this functions.  Personally, not a fan of functions that "check and then maybe do something" since they make the resulting code a bit harder to reason about

Reply:
My reason for making a separate function with a longer name is to reduce the time another developer spends trying to understand what the purpose is. I try to do this a lot, so if this is having the opposite effect then we should discuss it more in this thread.

Reply:
How about a name along the lines of:
`convertBoolStringToLuaAndMaybePush` or something?  Just to make sure that when someone reads the function call they know that it's not guaranteed that a value push to the stack will happen

Reply:
I changed it to checking what the type of the variable is, but removing the lua push so that there won't be uncertainty about whether or not a push to the lua stack was performed.